app/addons/documents/pagination/pagination.react.jsx

```diff
@@ -0,0 +1,198 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  "api",
+  "react",
+  'addons/documents/pagination/stores',
+  'addons/documents/pagination/actions',
+  ], function (FauxtonAPI, React, Stores, Actions) {
+    var indexPaginationStore = Stores.indexPaginationStore;
+
+    var IndexPaginationController = React.createClass({
+
+      getStoreState: function () {
+        return {
+          canShowPrevious: indexPaginationStore.canShowPrevious(),
+          canShowNext: indexPaginationStore.canShowNext(),
+        };
+      },
+
+      getInitialState: function () {
+        return this.getStoreState();
+      },
+
+      componentDidMount: function() {
+        indexPaginationStore.on('change', this.onChange, this);
+      },
+
+      componentWillUnmount: function() {
+        indexPaginationStore.off('change', this.onChange);
+      },
+
+      onChange: function () {
+        this.setState(this.getStoreState());
+      },
+
+      nextClicked: function (event) {
+        event.preventDefault();
+        event.stopPropagation();
+        if (!this.state.canShowNext) { return; }
+        Actions.paginateNext();
+      },
+
+      previousClicked: function (event) {
+        event.preventDefault();
+        event.stopPropagation();
+        if (!this.state.canShowPrevious) { return; }
+        Actions.paginatePrevious();
+      },
+
+      render: function () {
+        var canShowPreviousClassName;
+        var canShowNextClassName;
```

Comment:
would default them to an empty string, see https://issues.apache.org/jira/browse/COUCHDB-2577